app/policies/member_policy.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Base policy for member authorization
+class MemberPolicy < ApplicationPolicy
+  scope_for :relation do |relation, options|
+    namespace = options[:namespace]
+    direct_namespace_members = relation.where(namespace:).uniq(&:user_id)
```

Comment:
Instead of querying direct members and then querying indirect members we could just query it all at once:

e.g.

`relation.where(namespace_id: namespace_ids).order(access_level: :desc).uniq(&:user_id)`

Since `access_level` can only be increased down the hierarchy this should prefer direct members over indirect.

Reply:
Thanks for helping me clean up the queries! Updated in 0379505201abd0d8b0a317f46c1819e1c82f5c3e